staging/src/k8s.io/kube-aggregator/pkg/apiserver/openapi_aggregator.go

```diff
@@ -0,0 +1,253 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/emicklei/go-restful"
+	"github.com/go-openapi/spec"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/kube-openapi/pkg/aggregator"
+	"k8s.io/kube-openapi/pkg/builder"
+	"k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/pkg/handler"
+
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/kube-aggregator/pkg/apis/apiregistration"
+)
+
+const (
+	aggregatorUser = "system:aggregator"
+)
+
+type openAPIAggregator struct {
+	// Map of API Services' OpenAPI specs by their name
+	openAPISpecs map[string]*openAPISpecInfo
+
+	// provided for dynamic OpenAPI spec
+	openAPIService *handler.OpenAPIService
+
+	// Aggregator's OpenAPI spec.
+	aggregatorOpenAPISpec *spec.Swagger
+
+	// Local delegate's OpenAPI spec.
+	localDelegatesOpenAPISpec *spec.Swagger
+
+	contextMapper request.RequestContextMapper
+}
+
+func newOpenAPIAggregator(delegateHandler http.Handler, webServices []*restful.WebService, config *common.Config, pathHandler common.PathHandler, contextMapper request.RequestContextMapper) (s *openAPIAggregator, err error) {
+	s = &openAPIAggregator{
+		openAPISpecs:  map[string]*openAPISpecInfo{},
+		contextMapper: contextMapper,
+	}
+
+	// Get Local delegate's Spec
+	s.localDelegatesOpenAPISpec, err = s.loadOpenAPISpec(delegateHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build Aggregator's spec
+	s.aggregatorOpenAPISpec, err = builder.BuildOpenAPISpec(
+		webServices, config)
+	if err != nil {
+		return nil, err
+	}
+	// Remove any non-API endpoints from aggregator's spec. aggregatorOpenAPISpec
+	// is the source of truth for all non-api endpoints.
+	aggregator.FilterSpecByPaths(s.aggregatorOpenAPISpec, []string{"/apis/"})
+
+	// Build initial spec to serve.
+	specToServe, err := s.buildOpenAPISpec()
+	if err != nil {
+		return nil, err
+	}
+
+	// Install handler
+	s.openAPIService, err = handler.RegisterOpenAPIService(
+		specToServe, "/swagger.json", pathHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+// openAPISpecInfo is used to store OpenAPI spec with its priority.
+// It can be used to sort specs with their priorities.
+type openAPISpecInfo struct {
+	serviceName string
+	spec        *spec.Swagger
+	priority    int32
+}
+
+// byPriority can be used in sort.Sort to sort specs with their priorities.
+type byPriority []openAPISpecInfo
+
+func (a byPriority) Len() int      { return len(a) }
+func (a byPriority) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a byPriority) Less(i, j int) bool {
+	if a[i].priority != a[j].priority {
+		// Sort by priority, higher first
+		return a[i].priority > a[j].priority
+	} else {
```

Comment:
no else clause necessary. Just call return directly. That's more golang-idiomatic.